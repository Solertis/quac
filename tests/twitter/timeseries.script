#!/bin/bash
# Copyright (c) 2012-2013 Los Alamos National Security, LLC, and others.

# Compute keyword time series and related analyses.

. ./environment.sh

cp $QUACBASE/misc/halloween.xls $DATADIR
cd $DATADIR

exec 2>&1


## ngrams-build ##

# We use several partitions so that we can test that
# hash-based time series lookup works.
echo "$ ngrams-build ..."
ngrams-build --notimes --partitions 10 --min-occur 10 --run 4 --clean --jobdir series $TWEETDIR/pre > /dev/null

x ls -R series


## ngrams-search ##

# Test that ngrams-search complains properly if input is incorrect.
y 'ngrams-search --notimes --query candy /nonexistent || true'

# "candy" should peak near Halloween. (Note that midnight UTC is 8pm EST, so a
# lot of trick-or-treating related traffice will happen on November 1 UTC.)
x ngrams-search --query candy series/out
x ngrams-search --detail --query candy series/out

# "sandy" is the hurricane.
x ngrams-search --query sandy series/out
x ngrams-search --detail --query sandy series/out

# We include "a" because it's an n-gram that had trouble being sorted (see
# commit 885d3d)
x ngrams-search --query a series/out

# Make sure zeros are inserted properly: "taco" has no data on October 28
# (beginning), November 1 (middle), and November 4/5 (end).
x ngrams-search --detail --query taco series/out


## ngrams-correlate ##

# Test that ngrams-correlate complains properly if input is incorrect.
y "ngrams-correlate --notimes /nonexistent halloween.xls || true"
y "ngrams-correlate --notimes /bin/sh halloween.xls || true"
y "ngrams-correlate --notimes /bin /etc || true"
y "ngrams-correlate --notimes series /nonexistent.xls || true"

# Build the correlations. The sample rate is so tiny because:
#
#     0.01   spritzer 1%
#   * 0.05   5% sample for testing
#   * 0.015  1.5% of tweets are geotagged
#   * 0.5    last day of test sample is incomplete
#   -------
#   = 3.75e-6
#
echo "$ ngrams-correlate ..."
ngrams-correlate --notimes --min-similarity 0.7 --min-ppm 2000 --sample-rate 3.75e-6 --run 4 --clean --jobdir corr series/out halloween.xls > /dev/null

# Results
x ls -R corr
x head -10 corr/out/halloween:halloween.tsv
x head -10 corr/out/halloween:s%40ndy.tsv
