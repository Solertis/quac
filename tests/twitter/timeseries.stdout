$ ngrams-build ...
ngbld INFO     starting
ngbld INFO     loading metadata and setting up job
ngbld INFO     writing totals
ngbld INFO     running job
ngbld INFO     cleaning up job
ngbld INFO     done
$ ls -R series
series:
Makefile
out
tmp

series/out:
0
1
2
3
4
5
6
7
8
9
total.pkl.gz

series/tmp:
$ (ngrams-search --notimes --query candy /nonexistent || true)
ngsch FATAL    no input files in /nonexistent
$ ngrams-search --query candy series/out
candy	32	2012-10-28	2012-11-04
$ ngrams-search --detail --query candy series/out
2012-10-28	2	2302	868.81
2012-10-29	2	3086	648.088
2012-10-30	1	3055	327.332
2012-10-31	9	2950	3050.85
2012-11-01	10	2887	3463.8
2012-11-02	2	3078	649.773
2012-11-03	4	3014	1327.14
2012-11-04	2	3209	623.247
2012-11-05	0	1185	0.0
$ ngrams-search --query sandy series/out
sandy	119	2012-10-28	2012-11-04
$ ngrams-search --detail --query sandy series/out
2012-10-28	8	2302	3475.24
2012-10-29	45	3086	14582.0
2012-10-30	42	3055	13748.0
2012-10-31	8	2950	2711.86
2012-11-01	4	2887	1385.52
2012-11-02	5	3078	1624.43
2012-11-03	4	3014	1327.14
2012-11-04	3	3209	934.871
2012-11-05	0	1185	0.0
$ ngrams-search --query a series/out
a	3222	2012-10-28	2012-11-05
$ ngrams-search --detail --query taco series/out
2012-10-28	0	2302	0.0
2012-10-29	2	3086	648.088
2012-10-30	9	3055	2945.99
2012-10-31	3	2950	1016.95
2012-11-01	0	2887	0.0
2012-11-02	1	3078	324.886
2012-11-03	1	3014	331.785
2012-11-04	0	3209	0.0
2012-11-05	0	1185	0.0
$ (ngrams-correlate --notimes /nonexistent halloween.xls || true)
usage: ngrams-correlate [--min-similarity X] [--min-ppm X] [--sample-rate X]
                        [--run RUN] [--clean] [--map CMD] [--reduce CMD]
                        [--python CLASS] [--pyargs DICT] [--jobdir DIR]
                        [--partitions N] [--sortdir DIR] [--sortmem N]
                        [--update] [-h] [--notimes] [--unittest] [--verbose]
                        FILE [FILE ...]
ngrams-correlate: error: /nonexistent is not a directory
$ (ngrams-correlate --notimes /bin/sh halloween.xls || true)
usage: ngrams-correlate [--min-similarity X] [--min-ppm X] [--sample-rate X]
                        [--run RUN] [--clean] [--map CMD] [--reduce CMD]
                        [--python CLASS] [--pyargs DICT] [--jobdir DIR]
                        [--partitions N] [--sortdir DIR] [--sortmem N]
                        [--update] [-h] [--notimes] [--unittest] [--verbose]
                        FILE [FILE ...]
ngrams-correlate: error: /bin/sh is not a directory
$ (ngrams-correlate --notimes /bin /etc || true)
usage: ngrams-correlate [--min-similarity X] [--min-ppm X] [--sample-rate X]
                        [--run RUN] [--clean] [--map CMD] [--reduce CMD]
                        [--python CLASS] [--pyargs DICT] [--jobdir DIR]
                        [--partitions N] [--sortdir DIR] [--sortmem N]
                        [--update] [-h] [--notimes] [--unittest] [--verbose]
                        FILE [FILE ...]
ngrams-correlate: error: /etc is not a file
$ (ngrams-correlate --notimes series /nonexistent.xls || true)
usage: ngrams-correlate [--min-similarity X] [--min-ppm X] [--sample-rate X]
                        [--run RUN] [--clean] [--map CMD] [--reduce CMD]
                        [--python CLASS] [--pyargs DICT] [--jobdir DIR]
                        [--partitions N] [--sortdir DIR] [--sortmem N]
                        [--update] [-h] [--notimes] [--unittest] [--verbose]
                        FILE [FILE ...]
ngrams-correlate: error: /nonexistent.xls is not a file
$ ngrams-correlate ...
ngcor INFO     starting
ngcor INFO     setting up job
ngcor INFO     running job
ngcor INFO     cleaning up job
ngcor INFO     done
$ ls -R corr
corr:
Makefile
out
tmp

corr/out:
halloween:halloween.tsv
halloween:s%40ndy.tsv

corr/tmp:
$ head -10 corr/out/halloween:halloween.tsv
happyhalloween	0.998131075392	2078.28	0.0
happy halloween	0.996925606519	10847.5	0.0
trick or	0.995727657015	3728.81	0.0
thingsaboutme	0.993012003619	4067.8	0.0
trick	0.990344975957	3728.81	0.0
nba	0.9881173924	4582.65	0.0
or treating	0.9881173924	2372.88	0.0
treating	0.986329162782	2372.88	0.0
halloween	0.984747458302	28135.6	623.247
happy	0.983947838105	15593.2	5063.29
$ head -10 corr/out/halloween:s%40ndy.tsv
super	0.99810059828	3600.65	1558.12
sandy	0.99518702064	14582.0	934.871
hurricane	0.993076652371	8425.15	0.0
bonito	0.987933152078	2531.65	0.0
car	0.985450405022	2945.99	1303.21
storm	0.982598080446	2172.02	311.624
nba nba	0.980498061745	3273.32	0.0
nationaltextyourexday	0.980498001965	2291.33	0.0
day	0.980068437556	13748.0	7962.84
us	0.978356546112	6804.93	2531.65
